x-pack/plugins/global_search/public/services/search_service.ts

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { merge, Observable, timer, throwError } from 'rxjs';
+import { map, takeUntil } from 'rxjs/operators';
+import { duration } from 'moment';
+import { HttpStart, ApplicationStart } from 'src/core/public';
+import { GlobalSearchProviderResult } from '../../common/types';
+import { takeInArray } from '../../common/operators';
+import { processProviderResult } from '../../common/process_result';
+import { ILicenseChecker } from '../../common/license_checker';
+import { GlobalSearchResultProvider } from '../types';
+import { GlobalSearchClientConfigType } from '../config';
+import { GlobalSearchBatchedResults, GlobalSearchFindOptions } from './types';
+import { addNavigate, getDefaultPreference } from './utils';
+import { fetchServerResults } from './fetch_server_results';
+
+/** @public */
+export interface SearchServiceSetup {
+  registerResultProvider(provider: GlobalSearchResultProvider): void;
+}
+
+/** @public */
+export interface SearchServiceStart {
+  find(term: string, options: GlobalSearchFindOptions): Observable<GlobalSearchBatchedResults>;
+}
+
+interface SetupDeps {
+  config: GlobalSearchClientConfigType;
+  maxProviderResults?: number;
+  licenseChecker: ILicenseChecker;
+}
+
+interface StartDeps {
+  http: HttpStart;
+  application: ApplicationStart;
+}
+
+const defaultMaxProviderResults = 20;
+
+/** @internal */
+export class SearchService {
+  private readonly providers: GlobalSearchResultProvider[] = [];
+  private config?: GlobalSearchClientConfigType;
+  private http?: HttpStart;
+  private application?: ApplicationStart;
+  private maxProviderResults = defaultMaxProviderResults;
+  private licenseChecker?: ILicenseChecker;
+
+  setup({
+    config,
+    licenseChecker,
+    maxProviderResults = defaultMaxProviderResults,
+  }: SetupDeps): SearchServiceSetup {
+    this.config = config;
+    this.licenseChecker = licenseChecker;
+    this.maxProviderResults = maxProviderResults;
+
+    return {
+      registerResultProvider: (provider) => {
+        if (this.providers.map((p) => p.id).includes(provider.id)) {
+          throw new Error(`trying to register duplicate provider: ${provider.id}`);
+        }
+        this.providers.push(provider);
+      },
+    };
+  }
+
+  start({ http, application }: StartDeps): SearchServiceStart {
+    this.http = http;
+    this.application = application;
+
+    return {
+      find: (term, options) => this.performFind(term, options),
+    };
+  }
+
+  private performFind(term: string, options: GlobalSearchFindOptions) {
+    const licenseState = this.licenseChecker!.getState();
+    if (!licenseState.valid) {
+      return throwError(
+        `GlobalSearch API is disabled because of invalid license state: ${licenseState.message}`
```

Comment:
Not directly at least. this is an API error message, so it depends on the consumer's usage of it. But this may be more of a functional error than a technical one, so maybe translating it makes sense?

Reply:
> But this may be more of a functional error than a technical one, so maybe translating it makes sense?

Yeah, it makes sense to me

Reply:
done in https://github.com/elastic/kibana/pull/66293/commits/05d1372ee8b635a1705bb54709db4566838a179a